userspace/libscap/engine/source_plugin/source_plugin.c

```diff
@@ -229,6 +229,32 @@ static int32_t get_stats(struct scap_engine_handle engine, OUT scap_stats* stats
 	return SCAP_SUCCESS;
 }
 
+struct scap_stats_v2* get_source_plugin_stats_v2(struct scap_engine_handle engine, uint32_t flags, OUT uint32_t* nstats, OUT int32_t* rc)
+{
+	struct source_plugin_engine *handle = engine.m_handle;
+	*nstats = MAX_SOURCE_PLUGIN_COUNTERS_STATS;
+	scap_stats_v2* stats = (scap_stats_v2*)malloc(*nstats * sizeof(scap_stats_v2));
+
+	if (MAX_SOURCE_PLUGIN_COUNTERS_STATS > *nstats)
```

Comment:
No sure how this case can ever happen since we set the nstats value a couple of line above. Also applies for all the other engines.